### DISCUSSION TITLE
RemoteCertificateNameMismatch issue

### DISCUSSION THREAD

**Discussion by TheCriticalPath** [datalust/seq-tickets#1989] [Help]
I am receiving the following error on my local machine, in the Serilog self-log, when trying to initialize to the remote SEQ server and no logging occurs.

_The remote certificate is invalid according to the validation procedure RemoteCertificateNameMismatch_

I have deployed to my development server, the same server where SEQ is installed, and it appears to work fine.  

API keys are enabled, HTTP/s is required for ingestion.
The certificate is a wildcard certificate with multiple SAN entries.
SNI is enable on the server.

Can you help me resolve this issue so we can test the logging from local dev machines?

Local
MacOS Sonoma
VSCode, .netCore 7.0.12, Serilog:latest

Server 
Windows 2019 DC
Using HTTP.sys for SEQ

**Comment by nblumhardt** [comment]
Hi! Thanks for dropping us a line. This error is normally accurate - are you using a hostname in the Seq URI used to configure Serilog that would match the certificate name precisely?

One way to debug this is to take whatever URL you're configuring `WriteTo.Seq` with, including its scheme/host/port/path, and pasting that into a web browser. Does the browser accept the certificate without warnings?

If it's easier to share some details by email please feel free to drop us a line via `support@datalust.co`. HTH!

**Comment by TheCriticalPath** [comment]
Thanks for the response. 

My certificate is not an exact match to the SEQ url, but the browser does accept it without issue.

The CN for the certificate is *.&lt;orgname&gt;.com.  
The site is configured as:  d-seq.&lt;orgname&gt;.com.  

Again, the same configuration is successfully running on the server where SEQ is deployed, just not from local development machines.  Is there a provision for accessing the ServerCertificateValidationCallback to investigate the issue further?

**Reply by nblumhardt** [reply]
Hi @TheCriticalPath, thanks for your reply.

Yes, in the client application using Serilog, you can pass an `HttpMessageHandler` to the `WriteTo.Seq()` function and hook into the callback, e.g. as described in https://stackoverflow.com/a/46626858/138206.

My best bet is that the certificate issuer/CA cert isn't trusted on your production machines. It'd be great to hear how you go.

Best regards,.
Nick

**Reply by TheCriticalPath** [reply]
Hi @nblumhardt ,
    I was able to fix this on my MacOS development machine.  It appears that there is an issue with the site certificate not being standards compliant.  I had to evaluate the certificate in the Keychain Access app and use the Certificate Assistant to determine this error.
    
<img width="616" alt="image" src="https://github.com/datalust/seq-tickets/assets/6907305/f8ddff48-4c30-4223-b7d3-37b16cec6a0d">

To resolve this error locally, I was able to use custom security settings to trust this certificate for SSL.  I now need to reach out to the team that issues our internal certificates and figure out what is causing this to not be standards compliant.  In the meantime, I need to check out our windows development machines to see if I can do something similar there to bypass this issue.

The CA is fully trusted in both locations, but it appears that the site certificate has problems that the browser can handle, but Postman and the SEQ sink cannot.

Best Regards,
Tony

**Reply by nblumhardt** [reply]
Interesting - thanks for letting us know :+1:
